index.d.ts

```diff
@@ -18,9 +20,19 @@ declare module "react-css-themr"
 
 	}
 
+	interface ThemedComponent<P, S> extends React.Component<P, S>
+	{
+		getWrappedInstance(): React.Component<P, S>;
+	}
+
+	interface ThemedComponentClass<P, S> extends React.ComponentClass<P>
+	{
+		new(props?: P, context?: any): ThemedComponent<P, S>;
+	}
+
 	export function themr(
 		identifier: string,
 		defaultTheme?: {},
 		options?: IThemrOptions
-	);
+	): <P, S>(component: new() => React.Component<P, S>) => ThemedComponentClass<P, S>;
```

Comment:
still will not work if component accepts some props or context through constructor
```
consturctor(props: MyProps) {}
```
adding 
```
new(props?: P, context?: any)
```
will help.
After that for me it works perfect.